instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/ContextInstrumentation.java

```diff
@@ -6,65 +6,48 @@
 package io.opentelemetry.javaagent.instrumentation.opentelemetryapi;
 
 import static net.bytebuddy.matcher.ElementMatchers.isMethod;
-import static net.bytebuddy.matcher.ElementMatchers.isPublic;
 import static net.bytebuddy.matcher.ElementMatchers.isStatic;
 import static net.bytebuddy.matcher.ElementMatchers.named;
-import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
 
 import application.io.opentelemetry.context.Context;
-import application.io.opentelemetry.context.Scope;
+import application.io.opentelemetry.context.ContextStorage;
 import com.google.auto.service.AutoService;
-import io.opentelemetry.javaagent.instrumentation.api.ContextStore;
-import io.opentelemetry.javaagent.instrumentation.api.InstrumentationContext;
+import io.opentelemetry.javaagent.instrumentation.opentelemetryapi.context.AgentContextStorage;
 import io.opentelemetry.javaagent.tooling.Instrumenter;
-import java.util.HashMap;
+import java.util.Collections;
 import java.util.Map;
 import net.bytebuddy.asm.Advice;
 import net.bytebuddy.description.method.MethodDescription;
 import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.matcher.ElementMatcher;
 
+/**
+ * Returns {@link AgentContextStorage} as the implementation of {@link ContextStorage} in the
+ * application classpath. We do this instead of using the normal service loader mechanism to make
+ * sure there is no dependency on a system property or possibility of a user overriding this since
+ * it's required for instrumentation in the agent to work properly.
+ */
 @AutoService(Instrumenter.class)
 public class ContextInstrumentation extends AbstractInstrumentation {
 
   @Override
   public ElementMatcher<? super TypeDescription> typeMatcher() {
-    return named("application.io.opentelemetry.context.Context");
+    return named("application.io.opentelemetry.context.DefaultContext");
```

Comment:
We used to instrument the package-private class `TracingContextUtils`, now we do still end up with one instrumentation of `DefaultContext`